power_grid_model_c/power_grid_model/include/power_grid_model/component/fault.hpp

```diff
@@ -141,15 +157,25 @@ class Fault final : public Base {
     double x_f_;
 
     void check_sanity() const {
-        if (fault_type_ == FaultType::three_phase) {
-            if (fault_phase_ != FaultPhase::abc) {
-                throw InvalidShortCircuitPhases(fault_type_, fault_phase_);
-            }
-        }
-        else {
-            if (fault_phase_ == FaultPhase::abc) {
-                throw InvalidShortCircuitPhases(fault_type_, fault_phase_);
+        auto supported = [](FaultType fault_type) -> std::vector<FaultPhase> {
+            using enum FaultPhase;
+            switch (fault_type) {
+                case FaultType::three_phase:
+                    return {FaultPhase::nan, default_value, abc};
+                case FaultType::single_phase_to_ground:
+                    return {FaultPhase::nan, default_value, a, b, c};
+                case FaultType::two_phase:
+                    [[fallthrough]];
+                case FaultType::two_phase_to_ground:
+                    return {FaultPhase::nan, default_value, ab, ac, bc};
+                case FaultType::nan:
```

Comment:
same here, a `InvalidEnum` should be thrown.

Reply:
this one **is** checked during the update, so should i accept the `nan` value here or not?

Reply:
You are correct. The function is called in the constructor and the `fault_type` can be `nan` there. So we can keep this as is for this one.